Support parallel parsing for LineAsString input format

Avogar opened the pull request:
### Changelog category (leave one):
- Performance Improvement


### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support parallel parsing for LineAsString input format. This improves performance just slightly. This closes #42502.


> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

Avogar committed:
Support parallel parsing for LineAsString input format

davenger commented:
Looks like fileSegmentationEngineRegexpImpl function got renamed and moved into a separate file.
Could you please explain how this change enables parallel parsing? Am I missing something?

Avogar commented:
> Looks like fileSegmentationEngineRegexpImpl function got renamed and moved into a separate file. Could you please explain how this change enables parallel parsing? Am I missing something?

Oops, for some reason not all changes were uploaded, fixed it
